main.go

```diff
@@ -126,73 +109,266 @@ func (e *periodicExporter) Collect(ch chan<- prometheus.Metric) {
-			)
-			metricsChan <- prometheus.MustNewConstMetric(
-				memLimitDesc,
-				prometheus.GaugeValue,
-				float64(stat.Statistics.MemLimitBytes),
-				stat.Source, host, stat.FrameworkId,
-			)
-			metricsChan <- prometheus.MustNewConstMetric(
-				memRssDesc,
-				prometheus.GaugeValue,
-				float64(stat.Statistics.MemRssBytes),
-				stat.Source, host, stat.FrameworkId,
-			)
-		}
+func (e *periodicExporter) fetchNetworkStatistics(url *url.URL, host string, metricsChan chan <- prometheus.Metric) {
+	monitorURL := fmt.Sprintf("%s/monitor/statistics.json", url)
+	resp, err := httpClient.Get(monitorURL)
+	if err != nil {
+		log.Warn(err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	defer resp.Body.Close()
+
+	var stats []mesos_stats.Monitor
+	if err = json.NewDecoder(resp.Body).Decode(&stats); err != nil {
+		log.Warn("failed to deserialize response: ", err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	labels := []string{"task", "slave", "framework_id"}
+	for _, stat := range stats {
+		labelValues := []string{stat.Source, host, stat.FrameworkId}
+		e.gauge(metricsChan, labels, labelValues, "mesos_task_cpu_limit", float64(stat.Statistics.CpusLimit), "Fractional CPU limit.")
+		e.counter(metricsChan, labels, labelValues, "mesos_task_cpu_system_seconds_total", float64(stat.Statistics.CpusSystemTimeSecs), "Cumulative system CPU time in seconds.")
+		e.counter(metricsChan, labels, labelValues, "mesos_task_cpu_user_seconds_total", float64(stat.Statistics.CpusUserTimeSecs), "Cumulative system CPU time in seconds.")
+		e.gauge(metricsChan, labels, labelValues, "mesos_task_memory_limit_bytes", float64(stat.Statistics.MemLimitBytes), "Task memory limit in bytes.")
+		e.gauge(metricsChan, labels, labelValues, "mesos_task_memory_rss_bytes", float64(stat.Statistics.MemRssBytes), "Task memory RSS usage in bytes.")
 	}
 }
 
+func (e *periodicExporter) fetchSlaveMetricsSnapshot(url *url.URL, host string, metricsChan chan <- prometheus.Metric) {
+	monitorURL := fmt.Sprintf("%s/metrics/snapshot", url)
+	resp, err := httpClient.Get(monitorURL)
+	if err != nil {
+		log.Warn(err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	defer resp.Body.Close()
+
+	var stats mesos_metrics.SlaveMetrics
+	if err = json.NewDecoder(resp.Body).Decode(&stats); err != nil {
+		log.Warn("failed to deserialize response: ", err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	labels := []string{"node", "type"}
+	labelValues := []string{host, "slave"}
+	e.gauge(metricsChan, labels, labelValues, "mesos_cpus_percent", stats.CpuPercentage, "CPU percentage")
+	e.gauge(metricsChan, labels, labelValues, "mesos_cpus_total", float64(stats.CpuTotal), "CPU total")
+	e.gauge(metricsChan, labels, labelValues, "mesos_cpus_used", stats.CpuUsed, "CPU used")
+	e.gauge(metricsChan, labels, labelValues, "mesos_disk_percent", stats.DiskPercentage, "Disk percentage")
+	e.gauge(metricsChan, labels, labelValues, "mesos_disk_total", float64(stats.DiskTotal), "Disk total")
```

Comment:
mesos_cpus_total is the total cpu amount that Mesos can offer as resource. In the master metrics this is what's in total available to the cluster (addition of all slave values). In the slave metrics this is what's available to the slave.

mesos_disk_total is the same as mesos_cpu_total, but then for total disk space.
I made it a gauge, as these values can fluctuate